types/src/transaction/mod.rs

```diff
@@ -529,15 +674,153 @@ pub enum TransactionPayload {
     /// A multisig transaction that allows an owner of a multisig account to execute a pre-approved
     /// transaction as the multisig account.
     Multisig(Multisig),
+    /// A new transaction payload format with support for versioning.
+    /// Contains an executable (script/entry function) along with extra configuration.
+    /// Once this new format is fully rolled out, above payload variants will be deprecated.
+    Payload(TransactionPayloadInner),
+}
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub enum TransactionPayloadInner {
```

Comment:
is this the format we decided on previous design review?

Reply:
Yes. This is the format we decided.